### ISSUE TITLE
torch.flip not implemented for non-contiguous boolean tensors

### ISSUE THREAD

**Issue by szagoruyko** [pytorch/pytorch#52062]
## 🐛 Bug

Looks like flip_cuda does not support non-contiguous boolean tensors, and fails with not implemented error.

## To Reproduce

```python
import torch
a = torch.rand(8).cuda() > 0.5
torch.flip(a[::2], [0])
```

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-6550eb0d75ab> in <module>
----> 1 torch.flip(a[::2], [0])

RuntimeError: "flip_cuda" not implemented for 'Bool'
```

## Expected behavior

Should either support boolean flip like other tensors, or give a better error message.

## Environment

PyTorch 1.7.1+cu101

cc @izdeby @ngimel

**Comment by ngimel** [comment]
Fixed on master in #49895

**Comment by szagoruyko** [comment]
indeed, thanks Natalia! confirm that the issue is fixed in the nightlies, closing. Looking forward to new release.
